components/camel-kubernetes/src/main/java/org/apache/camel/component/kubernetes/customresources/KubernetesCustomResourcesProducer.java

```diff
@@ -0,0 +1,199 @@
+            }
+        }
+        LOG.info(customResourceJSON.toString());
+
+        MessageHelper.copyHeaders(exchange.getIn(), exchange.getOut(), true);
+        exchange.getOut().setBody(customResourceJSON);
+    }
+
+    protected void doDelete(Exchange exchange, String operation) throws Exception {
+        String customResourceName = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_CRD_INSTANCE_NAME, String.class);
+        String namespaceName = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_NAMESPACE_NAME, String.class);
+        if (ObjectHelper.isEmpty(customResourceName)) {
+            LOG.error("Delete a specific deployment require specify a deployment name");
+            throw new IllegalArgumentException("Delete a specific deployment require specify a deployment name");
+        }
+        if (ObjectHelper.isEmpty(namespaceName)) {
+            LOG.error("Delete a specific deployment require specify a namespace name");
+            throw new IllegalArgumentException("Delete a specific deployment require specify a namespace name");
+        }
+
+        JsonObject customResourceJSON = new JsonObject();
+        try {
+            customResourceJSON = new JsonObject(
+                    getEndpoint().getKubernetesClient().customResource(getCRDContext(exchange.getIn())).delete(namespaceName,
+                            customResourceName));
+        } catch (KubernetesClientException e) {
+            if (e.getCode() == 404) {
+                LOG.info("Custom resource instance not found", e);
+            } else {
+                throw e;
+            }
+        }
+
+        MessageHelper.copyHeaders(exchange.getIn(), exchange.getOut(), true);
+        exchange.getOut().setBody(customResourceJSON);
+    }
+
+    protected void doCreate(Exchange exchange, String operation) throws Exception {
+        String customResourceInstance = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_CRD_INSTANCE, String.class);
+        String namespaceName = exchange.getIn().getHeader(KubernetesConstants.KUBERNETES_NAMESPACE_NAME, String.class);
+
+        JsonObject gitHubSourceJSON = new JsonObject();
+        try {
+            gitHubSourceJSON = new JsonObject(
+                    getEndpoint().getKubernetesClient().customResource(getCRDContext(exchange.getIn())).create(namespaceName,
+                            customResourceInstance));
+        } catch (KubernetesClientException e) {
+            if (e.getCode() == 409) {
+                LOG.info("Custom resoure instance already exists", e);
+            } else {
+                throw e;
+            }
+        }
+        MessageHelper.copyHeaders(exchange.getIn(), exchange.getOut(), true);
+        exchange.getOut().setBody(gitHubSourceJSON);
+    }
+
+    private CustomResourceDefinitionContext getCRDContext(Message message) {
+        LOG.info("getCRDContext:" + message.getHeader(KubernetesConstants.KUBERNETES_CRD_NAME, String.class));
```

Comment:
No need for this log at info level. Too noisy. At least use debug level.

Reply:
Yes, it's too much.  I take them out.  Thanks!